pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -283,6 +286,16 @@ func NewContainerManager(mountUtil mount.Interface, cadvisorInterface cadvisor.I
 		qosContainerManager: qosContainerManager,
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.TopologyManager) {
+		klog.Infof("Node Config: %v", nodeConfig)
+		cm.topologyManager = topologymanager.NewManager(
+			nodeConfig.ExperimentalTopologyManagerPolicy,
+		)
+		klog.Infof("[topologymanager] Initilizing Topology Manager with %s policy", nodeConfig.ExperimentalTopologyManagerPolicy)
```

Comment:
this message could conflict if the policy name in the flag is invalid and `NewManager` falls back to the default policy

Reply:
I would argue that falling back to the default policy on an invalid policy name in the flag is bad behaviour. It should instead return an error, and fail fast in this case.

Reply:
i agree, this should be fatal to the kubelet, `NewManager` should return a `(Manager, err)`

Reply:
I think this was borrowed from the semantics of the CPUManager, which currently does a similar fallback to the None policy if an incorrect name is passed in the flag.

It's probably not worth blocking this PR until this is in place, since (at the moment) it's just potential logging error.

I will open a followup PR once this is merged to address this (in both the TopologyManager and the CPUManager).

Reply:
Going to resolve this conversation as there is a follow on PR planned. Thanks!

Reply:
Here is the PR: https://github.com/kubernetes/kubernetes/pull/80294

Depending on which one lands first (either this PR or that one), one of them will need to be updated to make the change that handles `NewManager()` returning an `error` at this point in the code.